src/pypistats/__init__.py

```diff
@@ -161,6 +161,10 @@ def pypi_stats_api(
         res["data"] = _monthly_total(res["data"])
     elif total == "all":
         res["data"] = _total(res["data"])
+    elif total is not None:
+        raise ValueError(
+            f"invalid value {total} provided for total: use 'all' or 'monthly'"
+        )
```

Comment:
Lint fix:
```suggestion
    elif total is not None:
    msg = f"invalid value {total} provided for total: use 'all' or 'monthly'"
        raise ValueError(msg)
```

It's to give slightly more friendly tracebacks by avoiding the error message being printed twice: https://docs.astral.sh/ruff/rules/f-string-in-exception/

Reply:
And I missed one yesterday, `total` can also be "daily": `pypistats overall pandas --daily`

https://github.com/hugovk/pypistats/blob/68c50b3e8f497ed5c99871e32bd989fdd154e1e8/src/pypistats/cli.py#L245

If we want to validate this, we can check its value at the top of the function.

Reply:
I like that linting rule. Ideally ruff would have flagged that for me automatically. Maybe this would be a separate PR, but it would be great to have docs for how to enable pre-commit for this repo. Would it make sense to `uv add ruff --dev` and the same with other dev dependencies, and then point devs to something like https://gist.github.com/zkurtz/4c61572b03e667a7596a607706463543 ?